### PR TITLE
Enforce deterministic brief roster reconciliation and payout alignment

### PR DESCRIPTION
**Pull request by whoabuddy**
## Summary
- enforce a deterministic 30-signal compile roster with explicit included-roster metadata in compile output and saved brief JSON
- add first-class `replaced` handling, keep `brief_included` compile-owned, and reconcile `brief_signals` plus unpaid brief-inclusion earnings on recompile
- add test-only compile/brief-signal helpers and regression coverage for over-cap, recompile, payout, review-status, and scoring edge cases

## Verification
- `npm test -- --run src/__tests__/brief-compile-reconciliation.test.ts src/__tests__/signal-review.test.ts src/__tests__/retraction.test.ts src/__tests__/scoring-math.test.ts`
- `npm test -- --run src/__tests__/brief-compile-reconciliation.test.ts src/__tests__/signal-review.test.ts`
- `npm run typecheck`
- `npx biome lint src/routes/signal-review.ts src/routes/signals.ts src/routes/brief-compile.ts`

## Notes
- left the unrelated local change in `public/index.html` out of this PR
- adjusted compile flow so brief persistence happens before payout side effects, avoiding new earnings without a matching saved brief artifact if save fails

### CODE REVIEW

File: src/routes/brief-compile.ts [hunk 1]
```diff
@@ -227,7 +247,24 @@ briefCompileRouter.post("/api/brief/compile", compileRateLimit, async (c) => {
...
+      }
+      const logger = c.get("logger");
+      logger.error("Failed to record brief signals", { date, error: briefSignalsResult.error });
+      briefSignalsWarning = `Failed to record brief_signals: ${briefSignalsResult.error ?? "unknown error"}. Signals remain in 'approved' state — retry compile to fix.`;
+    }
+  }
+
+  // Save the brief before payout side effects so a persistence failure does not
+  // leave new earnings behind without a matching saved artifact.
   const saveResult = await saveBrief(c.env, {
```
[the remark this answers is not in the document]
Comment (whoabuddy) [reply src/routes/brief-compile.ts]: I did not apply this suggestion as written. Saving the brief before `recordBriefSignals()` would let us persist a brief artifact whose roster reconciliation never happened, which is a worse contract break for this path because the saved brief would no longer reflect the active roster state. I kept roster reconciliation ahead of persistence so the saved artifact only lands after the selected roster has been applied, and I separately moved `saveBrief()` ahead of payout side effects plus made payout reconciliation failures return an error. The cleaner long-term fix is still a single DO endpoint/transaction for compile + reconcile + save.
